drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -2035,31 +2043,28 @@ static int rswitch_stop(struct net_device *ndev)
 	return 0;
 };
 
-static u32 rswitch_search_l3fwd(struct l3_ipv4_fwd_param *param)
-{
-	struct rswitch_private *priv = param->priv;
-
-	rs_write32(param->frame_type, priv->addr + FWLTHTS0);
-	rs_write32(0, priv->addr + FWLTHTS1);
-	rs_write32(0, priv->addr + FWLTHTS2);
-	rs_write32(param->src_ip, priv->addr + FWLTHTS3);
-	rs_write32(param->dst_ip, priv->addr + FWLTHTS4);
-
-	rswitch_reg_wait(priv->addr, FWLTHTSR0, LTHTS, 0);
-
-	return (rs_read32(priv->addr + FWLTHTSR0) & BIT(1));
-}
-
 bool is_l3_exist(struct rswitch_private *priv, u32 src_ip, u32 dst_ip)
 {
-	struct l3_ipv4_fwd_param param = {
-		.priv = priv,
-		.src_ip = src_ip,
-		.dst_ip = dst_ip,
-		.frame_type = LTHSLP0v4OTHER,
-	};
+	struct rswitch_device *rdev;
+	struct rswitch_ipv4_route *routing_list;
+	struct l3_ipv4_fwd_param_list *l3_param_list;
+
+	mutex_lock(&priv->rdev_list_lock);
+	mutex_lock(&priv->ipv4_forward_lock);
```

Comment:
It seems like this lock can be redundant because forward flag status has to be kept in mind only when you setup hw.